test/src/tests-consent.ts

```diff
@@ -0,0 +1,750 @@
+            .getInstance()
+            .Consent.createConsentState();
+        expect(consentState).to.be.ok;
+
+        consentState
+            .addGDPRConsentState('foo', (mParticle.Consent.createGDPRConsent(
+                true,
+                1
+            ) as unknown) as PrivacyConsentState)
+            .addGDPRConsentState('bar ', (mParticle.Consent.createGDPRConsent(
+                true,
+                2
+            ) as unknown) as PrivacyConsentState)
+            .addGDPRConsentState('BAZ ', (mParticle.Consent.createGDPRConsent(
+                false,
+                3
+            ) as unknown) as PrivacyConsentState)
+            .addGDPRConsentState('  ', (mParticle.Consent.createGDPRConsent(
+                false,
+                4
+            ) as unknown) as PrivacyConsentState);
+
+        const gdprConsentState = (consentState.getGDPRConsentState() as unknown) as GDPRConsentStateDictionary;
+
+        expect(gdprConsentState).to.have.property('foo');
+        expect(gdprConsentState).to.have.property('bar');
+        expect(gdprConsentState).to.have.property('baz');
+        expect(gdprConsentState).to.not.have.property('  ');
+        expect(gdprConsentState.foo).to.have.property('Timestamp', 1);
+        expect(gdprConsentState).to.not.have.property('bar ');
+        expect(gdprConsentState.bar).to.have.property('Timestamp', 2);
+        expect(gdprConsentState).to.not.have.property('BAZ ');
+        expect(gdprConsentState.baz).to.have.property('Timestamp', 3);
+
+        done();
+    });
+
+    it('Should normalize GDPR consent purposes on remove', done => {
+        const consentState = mParticle
+            .getInstance()
+            .Consent.createConsentState();
+        expect(consentState).to.be.ok;
+
+        consentState
+            .addGDPRConsentState('foo', (mParticle.Consent.createGDPRConsent(
+                true,
+                1
+            ) as unknown) as PrivacyConsentState)
+            .addGDPRConsentState('bar ', (mParticle.Consent.createGDPRConsent(
+                true,
+                2
+            ) as unknown) as PrivacyConsentState)
+            .addGDPRConsentState('BAZ ', (mParticle.Consent.createGDPRConsent(
+                false,
+                3
+            ) as unknown) as PrivacyConsentState);
+
+        const gdprConsentState = (consentState.getGDPRConsentState() as unknown) as GDPRConsentStateDictionary;
+
+        expect(gdprConsentState).to.have.property('foo');
+        expect(gdprConsentState).to.have.property('bar');
+        expect(gdprConsentState).to.have.property('baz');
+
+        consentState.removeGDPRConsentState('FOO');
+        consentState.removeGDPRConsentState('bAr   ');
+        consentState.removeGDPRConsentState(' bAz ');
+
+        // Fetch an updated version of the consent state to verify attributes were removed
+        const gdprConsentStateUpdate = (consentState.getGDPRConsentState() as unknown) as GDPRConsentStateDictionary;
```

Comment:
We may have to hop on a zoom to discuss this TS casting behavior and what potential solutions there are if any.